internal/controller/commitstatus_controller.go

```diff
@@ -89,22 +90,24 @@ func (r *CommitStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	commitStatusProvider, err := r.getCommitStatusProvider(ctx, cs)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, fmt.Errorf("failed to get CommitStatus provider: %w", err)
 	}
 
 	// We use retry on conflict to avoid conflicts when updating the status because so many other controllers will be
 	// creating and updating commit status and the API is very simple we try to avoid conflicts to update the status as
 	// soon as possible.
 	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		newCs := promoterv1alpha1.CommitStatus{}
-		err := r.Client.Get(ctx, req.NamespacedName, &newCs, &client.GetOptions{})
+		// TODO: consider skipping Get on the initial attempt. The object we already got might be up to date.
+		err = r.Client.Get(ctx, req.NamespacedName, &newCs, &client.GetOptions{})
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to get CommitStatus %q: %w", req.Name, err)
 		}
 
+		// TODO: consider pulling this outside the retry loop and instead using the reconcile requeue to handle SCM errors.
```

Comment:
I did this in the loop so because commit status are generally modified by other controllers and the status updates conflict so much that I wanted to not have to re-reconcile all the time. I originally had it outside this loop and it just ended up making noisy logs etc.

Reply:
This is one of those maybe document reasons in code things